echo_server.py

```diff
@@ -21,10 +26,74 @@ def echo_server():
                 if len(response) < 32:
                     done = True
                 text_received = "{}{}".format(text_received, response)
-            conn.sendall(text_received)
+            if text_received:
+                response_user = parase_request(text_received)
+            conn.sendall(response_user)
     except KeyboardInterrupt:
        server_socket.close()
 
+def response_ok(content, body_text):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: {}'.format(content)
+    body = body_text
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    response = [first_line, timestamp, header, size, body, end]
+    return end.join(response)
+
+def response_error(error_code, error_message):
+    first_line = 'HTTP/1.1 {} {}'.format(error_code, error_message)
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: text/plain'
+    body = '{} {}'.format(error_code, error_message)
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    error_response = [first_line, timestamp, header, size, body, end]
+    return end.join(error_response)
+
+def parase_request(client_request):
+    full_text = client_request.splitlines()
+    first_line = full_text[0].split(' ')
+    response = error_determination(first_line)
+    return response
+
+def error_determination(http_methods):
+
+    if http_methods[0] != 'GET':
+        error = response_error('405', 'Method Not Allowed')
+        return error
+    elif http_methods[2] != 'HTTP/1.1':
+        error = response_error('505', 'HTTP Version Not Supported')
+        return error
+    else:
+        return resolve_uri(http_methods[1])
+
+def resolve_uri(uri):
+    if os.path.isdir(os.path.abspath(uri)):
+        list_of_files = dir_list(uri)
+        response = response_ok('text/html', list_of_files)
+        return response
+    elif os.path.isfile(uri):
+        file_info = file_text(uri)
+        content = mimetypes.guess_type(uri)[0]
+        response = response_ok(content, file_info)
+        return response
+    else:
+        return response_error('404', 'Content Not Found')
+
+def file_text(uri):
+    file_information = open(uri, "r")
```

Comment:
This is also part of the problem.  the `r` mode for reading files assumes that the files are 'text' and makes some optimizations to the reading algorithm that change the data.  There is also a `binary` mode for reading that is called `rb`, that will **not** change the bytes as they are read, and your images will format correctly.